nixos/modules/services/networking/tedicross.nix

```diff
@@ -0,0 +1,142 @@
+              {
+                name = "Default bridge";
+                direction = "both";
+                telegram = {
+                  chatId = -123456789;
+                  relayJoinMessages = true;
+                  relayLeaveMessages = true;
+                  sendUsernames = true;
+                };
+                discord = {
+                  serverId = "DISCORD_SERVER_ID";
+                  channelId = "DISCORD_CHANNEL_ID";
+                  relayJoinMessages = true;
+                  relayLeaveMessages = true;
+                  sendUsernames = true;
+                  crossDeleteOnTelegram = true;
+                };
+              }
+            ];
+
+            debug = false;
+          }
+        '';
+        description = ''
+          <filename>settings.yaml</filename> configuration as a Nix attribute set.
+          Secret tokens should be specified using <option>environmentFile</option>
+          instead of this world-readable file.
+        '';
+      };
+
+      environmentFile = mkOption {
+        type = types.nullOr types.path;
+        default = null;
+        description = ''
+          File containing environment variables to be passed to the TediCross service,
+          in which secret tokens can be specified securely using the
+          <literal>TELEGRAM_BOT_TOKEN</literal> and <literal>DISCORD_BOT_TOKEN</literal>
+          keys.
+        '';
+      };
+
+      dataDir = mkOption {
+        type = types.str;
+        default = homeDir + "/data";
+        description = ''
+          Variable data directory of the TediCross service.
+        '';
+      };
+
+      user = mkOption {
+        type = types.string;
+        default = "tedicross";
+        description = ''
+          User to use for the TediCross service.
+        '';
+      };
+
+      group = mkOption {
+        type = types.string;
+        default = "tedicross";
+        description = ''
+          Group to use for the TediCross service.
+        '';
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    users.users = singleton {
+      description = "TediCross Telegram-Discord bridge user";
+      name = cfg.user;
+      group = cfg.group;
+      home = homeDir;
+      createHome = true;
+      isSystemUser = true;
+    };
+
+    users.groups = singleton {
+      name = cfg.group;
+    };
+
+    # from https://github.com/TediCross/TediCross/blob/master/guides/autostart/Linux.md
+    systemd.services.tedicross = {
+      description = "TediCross Telegram-Discord bridge service";
+      wantedBy = [ "multi-user.target" ];
+      wants = [ "network-online.target" ];
+      after = [ "network-online.target" ];
+      preStart = ''
+        mkdir -p '${cfg.dataDir}'
+        chmod 750 -R '${cfg.dataDir}'
+      '';
+      serviceConfig = {
+        Type = "simple";
+        ExecStart = "${pkgs.tedicross}/bin/tedicross --config='${configYAML}' --data-dir='${cfg.dataDir}'";
+        Restart = "always";
+        User = cfg.user;
```

Comment:
Without ever having used this program I can't say for sure, but I get the feeling [DynamicUser](https://www.freedesktop.org/software/systemd/man/systemd.exec.html) would be a better choice here. Unless you have specific reason not to I would entirely remove the `user` and `group` option in favour of DynamicUser.